addons/resource/models/resource_calendar.py

```diff
@@ -369,7 +369,10 @@ def _attendance_intervals_batch(self, start_dt, end_dt, resources=None, domain=N
             for resource in resources:
                 if resource and resource._is_flexible():
                 # If the resource is flexible, return the whole period from start_dt to end_dt with a dummy attendance
-                    dummy_attendance = self.env['resource.calendar.attendance']
+                    dummy_attendance = self.env['resource.calendar.attendance'].new({
+                        'duration_hours': (end - start).total_seconds() / 3600,
+                        'duration_days': (end - start).days,
```

Comment:
Shouldn't it be +1 ? Since a total time inferior to 24 hours will return 0 days 
```suggestion
                        'duration_days': (end - start).days + 1,
```

Reply:
Thanks for the review! I noticed that I did the same mistake for `closest_allocation_dict['days'] = (end_datetime - start_datetime).days` in hr_leave_type so I applied changes as well.